node-scripts/src/pipeline3/impl/dockerCompose.js

```diff
@@ -0,0 +1,249 @@
+            )
+            .toString() +
+          " build --pull" +
+          require("./dockerCompose").getInstanceServices(instanceDef) +
+          "\n"
+      );
+
+      script += scripts.remote(
+        instanceDef.deployment.host.value,
+        "sudo chown -R root:root " +
+          path
+            .resolve(instanceDef.deployment.hostDir, instanceDef.name)
+            .toString()
+      );
+
+      return script;
+    },
+    getDataScript: function(instanceDef) {
+      return "";
+    },
+    getArtifactsScript: function(instanceDef) {
+      return "";
+    }
+  },
+  startInstance: {
+    getDeploymentScript: function(instanceDef) {
+      let script = "";
+
+      script += scripts.remote(
+        instanceDef.deployment.host.value,
+        "cd " +
+          path
+            .resolve(
+              instanceDef.deployment.hostDir,
+              instanceDef.name,
+              "bahmni_docker"
+            )
+            .toString() +
+          " && " +
+          "docker-compose -p " +
+          instanceDef.name +
+          " --env-file=" +
+          path
+            .resolve(
+              instanceDef.deployment.hostDir,
+              instanceDef.name,
+              instanceDef.name + ".env"
+            )
+            .toString() +
+          " up -d" +
+          require("./dockerCompose").getInstanceServices(instanceDef) +
+          "\n"
+      );
+      return script;
+    },
+    getDataScript: function(instanceDef) {
+      return "";
+    },
+    getArtifactsScript: function(instanceDef) {
+      return "";
+    }
+  },
+  /**
+   * Util function that wraps the passed commands so each is applied either accordingly.
+   *
+   * @param {String} containerName - The name of the container.
+   * @param {String} ifExistsCommand - The command that should run if the container exists.
+   * @param {String} elseCommand - The command that will run if the container does *not* exist.
+   *
+   * @return {String} The script as a string.
+   */
+  ifExists: function() {},
+  restart: function(instanceDef, sudo) {
+    let script = "";
+    let path = require("path");
+    let distPath = path
+      .resolve(
+        instanceDef.deployment.hostDir,
+        instanceDef.name,
+        "bahmni_docker"
+      )
+      .toString();
+    script += "cd " + distPath + " && ";
+    if (sudo) {
+      script += "sudo ";
+    }
+    script += "docker-compose -p " + instanceDef.name + " restart ";
+
+    return script + "\n";
+  },
+  remove: function(instanceDef, sudo) {
+    let script = "";
+    let path = require("path");
+    let distPath = path
+      .resolve(
+        instanceDef.deployment.hostDir,
+        instanceDef.name,
+        "bahmni_docker"
+      )
+      .toString();
+    script += "cd " + distPath + " && ";
+    if (sudo) {
+      script += "sudo ";
+    }
+    script += "docker-compose -p " + instanceDef.name + " down -v ";
+
+    return script + "\n";
+  },
+  pull: function() {},
```

Comment:
Seems like this is not tested right? This should probably be unit tested in **dockerCompose.spec.js**